modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -1136,9 +1136,11 @@ private int moveMessagesFromDLCToNewDestination(List<Long> messageIds, String so
 
             messagesToRemove.add(metadata);
             // Update Andes message with all the chunk details
-            List<AndesMessagePart> messageParts = messageContent.get(messageId);
-            for (AndesMessagePart messagePart : messageParts) {
-                andesMessage.addMessagePart(messagePart);
+            if(!messageContent.isEmpty()) {
```

Comment:
NPE is thrown at : `List<AndesMessagePart> messageParts = messageContent.get(messageId);` before go inside loop.